src/joint_state_listener.cpp

```diff
@@ -171,8 +171,12 @@ int main(int argc, char** argv)
 
   // gets the location of the robot description on the parameter server
   urdf::Model model;
-  if (!model.initParam("robot_description"))
+  if (!model.initParam("robot_description")) {
+    if (model.links_.empty()) {
+      return 0;
+    }
     return -1;
```

Comment:
I know you didn't add this, but -1 (really 255 in bash) is considered a reserved exit code.  Replacing this with 1 would be better.

Reply:
https://github.com/ros/robot_state_publisher/blob/ff466b4ffec2e352664822a830ba795480d4edcf/src/joint_state_listener.cpp#L180
done :+1: also noticed line 180 is returning -1 so i modified it to match...